schema/test/unit/swrs_functions/refresh_swrs_version_data_test.sql

```diff
@@ -0,0 +1,189 @@
+update swrs.emission set report_id = 4 where id = 4;
+
+update swrs.fuel set report_id = 1 where id = 6;
+update swrs.fuel set report_id = 2 where id = 7;
+update swrs.fuel set report_id = 3 where id = 8;
+update swrs.fuel set report_id = 4 where id = 9;
+
+-- Create version 0 for applications 1,3,4
+insert into ggircs_portal.application_revision(application_id, version_number)
+  values (1, 0), (3, 0), (4, 0);
+insert into ggircs_portal.application_revision_status(application_id, version_number, application_revision_status)
+  values (1, 0, 'submitted'),(3, 0, 'submitted'),(4, 0, 'submitted');
+
+insert into ggircs_portal.form_result(application_id, version_number, form_id, form_result)
+  values
+    (1, 0, 1, '{}'),
+    (1, 0, 2, '{}'),
+    (1, 0, 3, '[{}]'),
+    (1, 0, 4, '[{}]'),
+    (3, 0, 1, '{}'),
+    (3, 0, 2, '{}'),
+    (3, 0, 3, '[{}]'),
+    (3, 0, 4, '[{}]'),
+    (4, 0, 1, '{}'),
+    (4, 0, 2, '{}'),
+    (4, 0, 3, '[{}]'),
+    (4, 0, 4, '[{}]');
+
+
+-- set different updated_at values for applications 1 & 2 (should update) / 3 & 4 (shouldn't update)
+update ggircs_portal.form_result set updated_at = now() - interval '3 days' where application_id < 3;
+update ggircs_portal.form_result set updated_at = now() - interval '1 day' where application_id >= 3;
+
+select test_helper.modify_triggers('enable');
+
+-- set all swrs_reports to be imported between updated_at values for applications 1 & 2 / 3 & 4
+update swrs.report set imported_at = now() - interval '2 days';
+update swrs.report set reporting_period_duration = (select reporting_year from ggircs_portal.opened_reporting_year());
+
+select is(
+  (select count(*) from ggircs_portal.form_result where application_id in (1,3,4) and version_number=0),
+  12::bigint,
+  'There are version 0 form results for applications 1, 3 & 4'
+);
+
+select is_empty(
+  $$
+    select * from ggircs_portal.form_result where application_id=2 and version_number=0
+  $$,
+  'There are no version 0 form_results for application with id = 2'
+);
+
+-- Call refresh function
+select ggircs_portal_private.refresh_swrs_version_data();
+
+select isnt_empty(
+  $$
+    select * from ggircs_portal.form_result where application_id=2 and version_number=0
+  $$,
+  'Refresh function created version 0 form_results for application with id = 2'
+);
+
+select is(
+  (select facility_name from ggircs_portal.ciip_admin where application_id=2 and version_number=0),
+  (select facility_name from ggircs_portal.get_swrs_facility_data(2, (select reporting_year from ggircs_portal.opened_reporting_year()))),
+  'Refresh function added facility data to the admin form for application with id = 2'
+);
+
+select is(
+  ((select updated_at from ggircs_portal.form_result where application_id=3 and version_number=0 and form_id=1) < now() - interval '12 hours'),
+  true::boolean,
+  'Application with id 3 was not updated by the refresh function because form_result.updated_by was after swrs.imported_at'
```

Comment:
```suggestion
  'Application with id 3 was not updated by the refresh function because form_result.updated_at was after swrs.imported_at'
```
same typo below